src/lib/formats/fs_hp98x5.cpp

```diff
@@ -0,0 +1,1547 @@
+	return accum;
+}
+
+void hp98x5_impl::split_string_n_dump(const std::vector<u8>& in, std::vector<u8>& out, unsigned bpr)
+{
+	auto it = in.cbegin();
+	auto out_ins = std::back_inserter(out);
+	u16 rec_type = REC_TYPE_1STSTR;
+
+	while (1) {
+		unsigned free_len = bpr - (out.size() % bpr);
+		if (free_len <= 4) {
+			// Not enough free space at end of current record: fill with EORs
+			while (free_len) {
+				w16b(out_ins, REC_TYPE_EOR);
+				free_len -= 2;
+			}
+		} else {
+			unsigned s_len = in.cend() - it;
+			unsigned s_part_len = std::min(free_len - 4 , s_len);
+			if (s_part_len == s_len) {
+				// Free space to EOR enough for what's left of string
+				break;
+			}
+			w16b(out_ins, rec_type);
+			w16b(out_ins, u16(s_len));
+			std::copy(it, it + s_part_len, out_ins);
+			it += s_part_len;
+			rec_type = REC_TYPE_MIDSTR;
+		}
+	}
+
+	w16b(out_ins, rec_type == REC_TYPE_MIDSTR ? REC_TYPE_ENDSTR : REC_TYPE_FULLSTR);
+	w16b(out_ins, u16(in.cend() - it));
+	std::copy(it, in.cend(), out_ins);
+	if (in.size() & 1) {
+		*out_ins = 0;
+	}
+}
+
+std::vector<u8> hp98x5_impl::convert_txt_2_data(const std::vector<u8>& in, unsigned bpr)
+{
+	std::vector<u8> res;
+
+	std::vector<u8> accum;
+	std::vector<u8> trailer;
+
+	// Split "in" into lines of text discarding trailing whitespace on each one
+	for (auto c : in) {
+		if (c == '\n') {
+			if (!accum.empty()) {
+				split_string_n_dump(accum, res, bpr);
+			}
+			accum.clear();
+			trailer.clear();
+		} else if (c == ' ' || c == '\n' || c == '\r' || c == '\t' || c == '\f' || c == '\v') {
+			trailer.push_back(c);
+		} else {
+			if (!trailer.empty()) {
+				accum.insert(accum.end(), trailer.cbegin(), trailer.cend());
+				trailer.clear();
+			}
+			accum.push_back(c);
+		}
+	}
+
+	// Fill free space of last record with EOFs
+	auto out_ins = std::back_inserter(res);
+	for (unsigned pad_len = bpr - (res.size() % bpr); pad_len; pad_len -= 2) {
+		w16b(out_ins, REC_TYPE_EOF);
+	}
+
+	return res;
+}
+
+const std::array<u16, hp98x5_impl::FS_COUNT> hp98x5_impl::m_fmt_interleave = {
+	// ********
+	// * 9825 *
+	// ********
+	2,
+	// ********
+	// * 9831 *
+	// ********
+	0,
+	// ********
+	// * 9845 *
+	// ********
+	7
+};
+
+const std::array<std::map<u8, std::string_view>, hp98x5_impl::FS_COUNT> hp98x5_impl::m_fs_type_maps = {
+	// ********
+	// * 9825 *
+	// ********
+	std::map<u8, std::string_view>{
+		{ 1, "DATA"sv },
+		{ 2, "PROG"sv },
+		{ 3, "KEYS"sv },
+		{ 5, "ALL"sv },
+		{ 6, "BPRG"sv }
+	},
+	// ********
+	// * 9831 *
+	// ********
+	std::map<u8, std::string_view>{
+		{ 1, "DATA"sv },
+		{ 9, "DATA"sv },
+		{ 18, "PROG"sv },
+		{ 19, "KEYS"sv },
+		{ 20, "NULL"sv },
+		{ 21, "MEMORY"sv },
+		{ 22, "BINARY"sv },
+		{ 23, "86DATA"sv }
+	},
+	// ********
+	// * 9845 *
+	// ********
+	std::map<u8, std::string_view>{
+		{ 0, "BKUP"sv },
+		{ 8, "DATA"sv },
+		{ 16,"PROG"sv },
+		{ 24,"KEYS"sv },
+		{ 32,"BDAT"sv },
+		{ 40,"ALL"sv  },
+		{ 48,"BPRG"sv },
+		{ 56,"OPRM"sv },
+		{ 60,"ASMB"sv }
+	}
+};
```

Comment:
Static `std::map` variables require runtime initialisation and hence hurts startup performance whether this filesystem module is used or not.  Can you either use arrays instead and search them using `std::lower_bound` (will give the same O(log2(n)) complexity as a `std::map` with better locality), or initialise them when the filesystem module is actually used?